config_bucket.go

```diff
@@ -11,32 +11,45 @@ import (
 )
 
 const (
-	BucketTypeLocal = "local"
-	BucketTypeS3    = "s3"
+	BucketTypeGridFS = "gridfs"
+	BucketTypeLocal  = "local"
+	BucketTypeS3     = "s3"
 )
 
-// BucketConfig represents the admin config section for bucket storage.
+// BucketsConfig represents the admin config section for bucket storage.
+type BucketsConfig struct {
+	LogBucket BucketConfig `bson:"log_bucket" json:"log_bucket" yaml:"log_bucket"`
+}
+
+var bucketsConfigLogBucketKey = bsonutil.MustHaveTag(BucketsConfig{}, "LogBucket")
+
+// BucketConfig represents the admin config for an individual bucket.
 type BucketConfig struct {
-	LogBucket Bucket `bson:"log_bucket" json:"log_bucket" yaml:"log_bucket"`
+	Name   string `bson:"name" json:"name" yaml:"name"`
+	Type   string `bson:"type" json:"type" yaml:"type"`
```

Comment:
nit: if this only allows the `BucketType` constants to be set here, then consider not using a plain string and making a `BucketType` enum + making those constants of that type.